backend/service/token.go

```diff
@@ -1,28 +1,40 @@
 package service
 
 import (
+	"context"
 	"log"
 	"time"
 
 	"aidanwoods.dev/go-paseto"
+	errors "github.com/pkg/errors"
 )
 
 type TokenIFace interface {
 	NewToken(string) (string, error)
 	ValidateToken(string) (bool, error)
 	GetUidFromToken(string) (string, error)
-	// InvalidateToken(string) error
+	GenerateAccessToken(string) (string, error)
+	GenerateRefreshToken(string) (string, error)
+	RefreshAccessToken(string) (string, string, error)
+	InvalidateToken(string) error
 }
 
+const (
+	AccessTokenLife  = 5 * time.Minute
+	RefreshTokenLife = 7 * 24 * time.Hour
+)
+
 type Token struct {
 	key *paseto.V4AsymmetricSecretKey
+	Redis RedisIFace
 }
 
-func NewTokenServicer() *Token {
+func NewTokenServicer(redis RedisIFace) *Token {
```

Comment:
Maybe we should generalize "RedisIFace" to "InMemoryDbIFace". Not needed, but would distance the technology from the data structure. This would follow the `DbIFace` convention from before.